src/utils/glue.h

```diff
@@ -211,6 +217,21 @@ struct funcbox : box {
     matryoshka_t args;
 };
 
+template<typename Klazz>
+struct getter : box {
```

Comment:
maybe call it different than "getter" as there are so many getters everywhere) maybe smth like "applicable" or "cast_and_apply" as it mainly casts the object from the first argument and applies the given function to it